src/components/bcros/filing/CommonTemplate.vue

```diff
@@ -109,9 +110,16 @@ const isStatusApproved = computed(() => isFilingStatus(filing.value, FilingStatu
 
 const isShowBody = ref(false)
 
-const showDetails = () => {
+const showDetails = async () => {
   if (filing.value.documents === undefined && filing.value.documentsLink) {
-    loadDocumentList(filing.value)
+    ui.fetchingData = true
+
+    await loadDocumentList(filing.value)
+
+    // wait for another 500ms to show the loading modal
+    await new Promise(resolve => setTimeout(resolve, 250))
```

Comment:
I recommend moving this to a common util file, and name it "sleep" to be consistent with other UIs.